### PR TITLE
NXP-29129: refresh aws s3 token

### PR DESCRIPTION
**Pull request by nelsonsilva**

### CODE REVIEW

File: addons/amazon-s3-online-storage/index.js [hunk 1]
```diff
@@ -95,14 +95,28 @@ class S3Provider {
...
+    const credentials = new AWS.Credentials(
+      options.awsSecretKeyId,
+      options.awsSecretAccessKey,
+      options.awsSessionToken,
+    );
+    credentials.expireTime = new Date(options.expiration);
+    credentials.refresh = (cb) =>
+      this._resource(`upload/${this.batchId}/refreshToken`)
+        .post()
+        .then((response) => {
```
Comment (nelsonsilva) [inline addons/amazon-s3-online-storage/index.js:107]: I'll add a `catch(cb)` before merging this

File: addons/amazon-s3-online-storage/index.js [hunk 2]
```diff
@@ -95,14 +95,28 @@ class S3Provider {
   }
 
   _initCredentials(options) {
+    const credentials = new AWS.Credentials(
+      options.awsSecretKeyId,
+      options.awsSecretAccessKey,
+      options.awsSessionToken,
+    );
+    credentials.expireTime = new Date(options.expiration);
+    credentials.refresh = (cb) =>
```
Comment (Gabez0r) [inline addons/amazon-s3-online-storage/index.js:104]: When will this `refresh` be called? Do we need to implement `needsRefresh` as well?

Comment (nelsonsilva) [reply addons/amazon-s3-online-storage/index.js:104]: No, default `needsRefresh` implementaton works with the `expiration` info, see https://github.com/aws/aws-sdk-js/blob/master/lib/credentials.js#L95

Comment (Gabez0r) [reply addons/amazon-s3-online-storage/index.js:104]: :ok_hand:

**Pull request closed without merging**